Emby.Server.Implementations/LiveTv/EmbyTV/EmbyTV.cs

```diff
@@ -1776,44 +1730,16 @@ private async Task SaveRecordingImage(string recordingPath, LiveTvProgram progra
                 image = await _libraryManager.ConvertImageToLocal(program, image, 0).ConfigureAwait(false);
             }
 
-            string imageSaveFilenameWithoutExtension = null;
-
-            switch (image.Type)
+            string imageSaveFilenameWithoutExtension = image.Type switch
             {
-                case ImageType.Primary:
-
-                    if (program.IsSeries)
-                    {
-                        imageSaveFilenameWithoutExtension = Path.GetFileNameWithoutExtension(recordingPath) + "-thumb";
-                    }
-                    else
-                    {
-                        imageSaveFilenameWithoutExtension = "poster";
-                    }
-
-                    break;
-                case ImageType.Logo:
-                    imageSaveFilenameWithoutExtension = "logo";
-                    break;
-                case ImageType.Thumb:
-                    if (program.IsSeries)
-                    {
-                        imageSaveFilenameWithoutExtension = Path.GetFileNameWithoutExtension(recordingPath) + "-thumb";
-                    }
-                    else
-                    {
-                        imageSaveFilenameWithoutExtension = "landscape";
-                    }
-
-                    break;
-                case ImageType.Backdrop:
-                    imageSaveFilenameWithoutExtension = "fanart";
-                    break;
-                default:
-                    break;
-            }
+                ImageType.Primary => program.IsSeries ? Path.GetFileNameWithoutExtension(recordingPath) + "-thumb" : "poster",
+                ImageType.Logo => "logo",
+                ImageType.Thumb => program.IsSeries ? Path.GetFileNameWithoutExtension(recordingPath) + "-thumb" : "landscape",
+                ImageType.Backdrop => "fanart",
+                _ => null
+            };
 
-            if (string.IsNullOrWhiteSpace(imageSaveFilenameWithoutExtension))
+            if (imageSaveFilenameWithoutExtension == null)
```

Comment:
why?

Reply:
It can't possibly be empty or whitespace only